transport/tls-record-frag/writer.go

```diff
@@ -0,0 +1,105 @@
+package tlsrecordfrag
+
+import (
+	"errors"
+	"io"
+)
+
+type tlsRecordFragWriter struct {
+	writer      io.Writer
+	prefixBytes uint32
+}
+
+const maxRecordLength = 16384
+
+func NewWriter(writer io.Writer, prefixBytes uint32) *tlsRecordFragWriter {
+	return &tlsRecordFragWriter{writer, prefixBytes}
+}
+
+func (w *tlsRecordFragWriter) dontFrag(first []byte, source io.Reader) (written int64, err error) {
+	tmp, err := w.writer.Write(first)
+	written = int64(tmp)
+	w.prefixBytes = 0
+	if err != nil {
+		return written, err
+	}
+	n, err := io.Copy(w.writer, source)
+	written += n
+	return written, err
+}
+
+func (w *tlsRecordFragWriter) ReadFrom(source io.Reader) (written int64, err error) {
+	if 0 < w.prefixBytes {
+		var first [5]byte
```

Comment:
Let's give it a clearer name.
```suggestion
		var recordHeader [5]byte
```